src/ResourceManagement/Network/VirtualNetworkGatewayConnectionImpl.cs

```diff
@@ -274,16 +296,45 @@ public IReadOnlyCollection<TunnelConnectionHealth> TunnelConnectionStatus()
             return new ReadOnlyCollection<TunnelConnectionHealth>(Inner.TunnelConnectionStatus);
         }
 
-        public IWithCreate WithTags(IDictionary<string, string> tags)
+//        public VirtualNetworkGatewayConnectionImpl WithTags(IDictionary<string, string> tags)
+//        {
+//            base.WithTags(tags);
+//            return this;
+//        }
+//
+//        public VirtualNetworkGatewayConnectionImpl WithTag(string key, string value)
+//        {
+//            base.WithTag(key, value);
+//            return this;
+//        }
+        UpdatableWithTags.UpdatableWithTags.IUpdateWithTags<IVirtualNetworkGatewayConnection> IUpdatableWithTags<IVirtualNetworkGatewayConnection>.UpdateTags()
         {
-            base.WithTags(tags);
-            return this;
+            return UpdateTags();
         }
 
-        public IWithCreate WithTag(string key, string value)
+        IWithCreate IDefinitionWithTags<IWithCreate>.WithTags(IDictionary<string, string> tags)
         {
-            base.WithTag(key, value);
-            return this;
+            throw new System.NotImplementedException();
+        }
+
+        IUpdate IUpdateWithTags<IUpdate>.WithTag(string key, string value)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        IUpdate IUpdateWithTags<IUpdate>.WithoutTag(string key)
+        {
+            throw new System.NotImplementedException();
```

Comment:
>NotImplementedException [](start = 29, length = 23)

If this is by design then change all not implemented exceptions to not supported exception.